pkg/graphql/execution_engine_v2.go

```diff
@@ -199,6 +199,12 @@ func WithAdditionalHttpHeaders(headers http.Header, excludeByKeys ...string) Exe
 	}
 }
 
+func WithHeaderModifier(modifier postprocess.HeaderModifier) ExecutionOptionsV2 {
```

Comment:
can you add an early return to the inner function, so that it doesn't add the new post-processor if `modifier` is `nil`